pkg/util/net/net.go

```diff
@@ -72,3 +89,163 @@ func setBuffers(rc syscall.RawConn, sz int) error {
 
 	return err
 }
+
+func RemovePrivateDNSZone(
+	ctx context.Context,
+	log *logrus.Entry,
+	privateZonesClient privatedns.PrivateZonesClient,
+	configcli configclient.Interface,
+	mcocli mcoclient.Interface,
+	kubernetescli kubernetes.Interface,
+	vNetLinksClient privatedns.VirtualNetworkLinksClient,
+	resourceGroupID string) error {
+	if privateZonesClient == nil {
+		return errors.New("privateZonesClient is nil")
+	}
+
+	resourceGroup := stringutils.LastTokenByte(resourceGroupID, '/')
+	zones, err := privateZonesClient.ListByResourceGroup(ctx, resourceGroup, nil)
```

Comment:
Elsewhere you're verbose enough to specify that these are private zones, like in RemoveZones. Can we do that here too and make this variable `privateZones`?